bridger/go_explore_phase_1.py

```diff
@@ -32,17 +63,17 @@ def update_times_since_led_to_something_new(self, state, led_to_something_to_new
             self._cache[key].steps_since_led_to_something_new = 0
         else:
             self._cache[key].steps_since_led_to_something_new += 1
-    
-    def visit(self, state, trajectory: list[int], reward: float) -> bool:
+
+    def visit(self, state, trajectory: tuple[int], rewards: tuple[float]) -> bool:
         key = hash_utils.hash_tensor(state)
         if key in self._cache:
             entry = self._cache[key]
             entry.visit_count += 1
-            if reward > entry.reward or (reward == entry.reward and len(trajectory) < len(entry.trajectory)):
-                entry.reward = reward
+            if sum(rewards) > sum(entry.rewards) or (sum(rewards) == sum(entry.rewards) and len(trajectory) < len(entry.trajectory)):
```

Comment:
Our rewards are negative so you need to switch the < sign. I did this with the backward algorithm too